lib/atscppapi/src/utils_internal.cc

```diff
@@ -99,6 +97,8 @@ handleTransactionEvents(TSCont cont, TSEvent event, void *edata)
 void
 setupTransactionManagement()
 {
+  // Reserve a transaction slot
+  TSAssert(TS_SUCCESS == TSHttpArgIndexReserve("atscppapi", "ATS CPP API", &TRANSACTION_STORAGE_INDEX));
```

Comment:
Hmm, OK I guess this is safe because of the ``pthread_once`` in ``utils::internal::initTransactionManagement()``. It seems a little fragile to me though. I think it would be more obvious if this was just done in ``GlobalPlugin::GlobalPlugin()`` and the intermediate functions were removed.